crates/typst/src/math/stretch.rs

```diff
@@ -1,12 +1,151 @@
+/// regard.
+///
+/// ```example
+/// $ H stretch(=)^"define" U + p V $
+/// $ f : X stretch(->>, size: #150%)_"surjective" Y $
+/// $ x stretch(harpoons.ltrb, size: #3em) y
+///     stretch(\[, size: #150%) z $
+/// ```
+#[elem(LayoutMath)]
+pub struct StretchElem {
+    /// The glyph to stretch.
+    #[required]
+    pub body: Content,
+
+    /// The size to stretch to, relative to the glyph's current size.
+    pub size: Smart<Rel<Length>>,
+
+    /// Whether the size should be relative to the width of any top or bottom
+    /// attachments.
+    ///
+    /// ```example
+    /// #set math.stretch(attach: false)
+    /// $ stretch(->, size: #250%)_"very long text" $
+    /// ```
+    #[default(true)]
+    pub attach: bool,
+}
+
+impl LayoutMath for Packed<StretchElem> {
+    #[typst_macros::time(name = "math.stretch", span = self.span())]
+    fn layout_math(&self, ctx: &mut MathContext, styles: StyleChain) -> SourceResult<()> {
+        let mut fragment = ctx.layout_into_fragment(self.body(), styles)?;
+        stretch_fragment(
+            ctx,
+            styles,
+            &mut fragment,
+            None,
+            None,
+            self.size(styles),
+            Abs::zero(),
+        );
+        ctx.push(fragment);
+        Ok(())
+    }
+}
+
+/// Attempts to stretch the given fragment by/to the amount given in stretch.
+pub(super) fn stretch_fragment(
+    ctx: &mut MathContext,
+    styles: StyleChain,
+    fragment: &mut MathFragment,
+    axis: Option<Axis>,
+    relative_to: Option<Abs>,
+    stretch: Smart<Rel<Length>>,
+    short_fall: Abs,
+) {
+    let glyph = match fragment {
+        MathFragment::Glyph(glyph) => glyph.clone(),
+        MathFragment::Variant(variant) => {
+            GlyphFragment::new(ctx, styles, variant.c, variant.span)
+        }
+        _ => return,
+    };
+
+    let axis = if let Some(stretch_axis) = axis {
+        stretch_axis
+    } else if let Some(stretch_axis) = is_stretchable(ctx, &glyph) {
+        stretch_axis
+    } else {
+        return;
+    };
+
+    let relative_to_size = relative_to.unwrap_or_else(|| fragment.size().get(axis));
+
+    let mut variant = stretch_glyph(
+        ctx,
+        glyph,
+        stretch
+            .unwrap_or(Rel::one())
+            .resolve(styles)
+            .relative_to(relative_to_size),
+        short_fall,
+        axis,
+    );
+
+    if axis == Axis::Y {
+        variant.align_on_axis(ctx, delimiter_alignment(variant.c));
+    }
+
+    *fragment = MathFragment::Variant(variant);
+}
+
+pub(super) fn delimiter_alignment(delimiter: char) -> VAlignment {
+    match delimiter {
+        '\u{231c}' | '\u{231d}' => VAlignment::Top,
+        '\u{231e}' | '\u{231f}' => VAlignment::Bottom,
+        _ => VAlignment::Horizon,
+    }
+}
+
+/// Return whether the glyph is stretchable and if it is, along which axis it
+/// can be stretched.
+fn is_stretchable(ctx: &MathContext, base: &GlyphFragment) -> Option<Axis> {
```

Comment:
I think the name should be changed since it doesn't return a boolean. Maybe `stretch_axis`